src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1261 @@
+
+    {
+        constexpr beman::optional26::optional<int&> o;
+        EXPECT_TRUE(!o);
+
+        constexpr beman::optional26::optional<int&> oo = o;
+        EXPECT_TRUE(!oo);
+    }
+
+    {
+        constexpr static auto                              i = 42;
+        constexpr beman::optional26::optional<const int&> o = i;
+        EXPECT_TRUE(o);
+        EXPECT_TRUE(*o == 42);
+
+        constexpr beman::optional26::optional<const int&> oo = o;
+        EXPECT_TRUE(oo);
+        EXPECT_TRUE(*oo == 42);
+    }
+
+}
+
+TEST(OptionalConstexprTest, Constructors3) {
+    constexpr beman::optional26::optional<int> ie;
+    constexpr beman::optional26::optional<int> i4 = ie;
+    EXPECT_FALSE(i4);
+
+    using beman::optional26::tests::base;
+    using beman::optional26::tests::derived;
+
+    constexpr base                              b{1};
+    constexpr derived                           d(1, 2);
+    constexpr beman::optional26::optional<base> b1{b};
+    constexpr beman::optional26::optional<base> b2{d};
+
+    constexpr beman::optional26::optional<derived> d2{d};
+    constexpr beman::optional26::optional<base>    b3 = d2;
+    constexpr beman::optional26::optional<base>    b4{d2};
+    std::ignore = b1;
+    std::ignore = b2;
+    std::ignore = b3;
+    std::ignore = b4;
+    std::ignore = d2;
+
+}
+
+namespace {
+class NoDefault {
+    int v_;
+
+  public:
+    constexpr NoDefault(int v) : v_(v) {}
+};
+} // namespace
+
+TEST(OptionalConstexprTest, NonDefaultConstruct) {
+    constexpr NoDefault                              i = 7;
+    constexpr beman::optional26::optional<NoDefault> v1{};
+    constexpr beman::optional26::optional<NoDefault> v2{i};
+    std::ignore = v1;
+    std::ignore = v2;
+}
+
+consteval bool testConstexprAssignmentValue() {
+    bool retval = true;
+    beman::optional26::optional<int> o1 = 42;
+    beman::optional26::optional<int> o2 = 12;
+    beman::optional26::optional<int> o3;
+
+    o1 = o1;
+    retval &= (*o1 == 42);
+
+    o1 = o2;
+    retval &= (*o1 == 12);
+
+    o1 = o3;
+    retval &= (!o1);
+
+    o1 = 42;
+    retval &= (*o1 == 42);
+
+    o1 = beman::optional26::nullopt;
+    retval &= (!o1);
+
+    o1 = std::move(o2);
+    retval &= (*o1 == 12);
+
+    beman::optional26::optional<short> o4 = 42;
+
+    o1 = o4;
+    retval &= (*o1 == 42);
+
+    o1 = std::move(o4);
+    retval &= (*o1 == 42);
+
+    /*
+      template <class U = T>
+      constexpr optional& operator=(U&& u)
+    */
+    short s = 54;
+    o1      = s;
+    retval &= (*o1 == 54);
+
+    struct not_trivial_copy_assignable {
+        int i_;
+        constexpr not_trivial_copy_assignable(int i) : i_(i) {}
+        constexpr not_trivial_copy_assignable(const not_trivial_copy_assignable&) = default;
+        constexpr not_trivial_copy_assignable& operator=(const not_trivial_copy_assignable& rhs) {
+            i_ = rhs.i_;
+            return *this;
+        }
+    };
+
+    /*
+      optional& operator=(const optional& rhs)
```

Comment:
What do you think to always use `//` multi line comment?

Reply:
That's fine with me.